code/LeftAndMain.php

```diff
@@ -1755,6 +1755,21 @@ public function CMSVersionNumber()
         return $lockModules[$moduleName];
     }
 
+    /**
+     * Returns the minor version e.g. 4.10
+     * Will handle 4.10.x-dev by removing .x-dev
+     *
+     * @return string
+     */
+    public function CMSMinorVersionNumber(): string
+    {
+        $version = $this->CMSVersionNumber();
+        if (preg_match('#^([1-9])\.([1-9][0-9]?)#', $version, $m)) {
```

Comment:
This will have the same problem as we have now if we got to three-digit versions — unlikely but not impossible.

I'm also not 100% sure we should display the same thing for stable and dev versions, e.g. 4.10 for both 4.10 and 4.10.x-dev. Could the hover show the full thing?

Reply:
I'm with Michal on supporting 3-digit minor version. We'll probably never get there, but the regex for supporting a version number like 123.456.789 is simpler to understand any way.

```suggestion
        if (preg_match('#^(\d+)\.(\d+)#', $version, $m)) {
```

There could be some interesting value in adding some bit more detail about the stability of what you have installed ... if only for our own sake. However, I think that goes beyond the scope of this card which is just to make sure we don't confuse people when they upgrade to 4.10. We have a related card to convert the help menu to React and implement some updated "Silverstripe CMS" branding. I think that discussion would make more sense there.

Reply:
Fair enough re: the stability aspect and more info, happy to leave that for the other card.

Since `#^(\d+)\.(\d+)#` isn't used for validation but rather just extracting the major and minor version, knowing that the version is a valid composer version, this simplified approach is fine. Using the [official semver regex](https://semver.org/#is-there-a-suggested-regular-expression-regex-to-check-a-semver-string) would be an overkill here.